health/health_log.c

```diff
@@ -195,9 +195,26 @@ static inline ssize_t health_alarm_log_read(RRDHOST *host, FILE *fp, const char
         host->health_log_entries_written++;
         line++;
 
-        int max_entries = 33, entries = 0;
+        int max_entries = 33, entries = 0, non_ascii = 0;
         char *pointers[max_entries];
 
+        char *c = buf;
+        while (*c) {
+            if (unlikely(!isascii(*c))) {
+                error(
+                    "HEALTH [%s]: line %zu of file '%s' contains a non-ascii character, ignoring.",
+                    host->hostname,
+                    line,
+                    filename);
+                non_ascii = 1;
+                break;
+            }
+            c++;
+        }
+
+        if (unlikely(non_ascii == 1))
```

Comment:
Considering that `*c` will have `NULL` when success cases happen, I think we could remove the `non_ascii` variable.

Reply:
Ah, yes, you are of course correct! Pushed it in https://github.com/netdata/netdata/pull/11193/commits/783ebf79d2857e6411202cad55c7d0c835af5147